### PR TITLE
Primitive types

### PR DESCRIPTION
**Pull request by josh11b**
A few changes:
- Merge content from the `primitive_types.md` design doc into the overall design `README.md` since there was so much overlap and no need for two copies.
- Consistently spell integer types `Carbon.Int(N)` and `Carbon.UInt(N)`, including the `Carbon.` prefix and avoiding `Unsigned(N)`.
- Consistently use a comprehensive set of floating-point types.
- Incorporates #2015 into the design docs.

### CODE REVIEW

File: docs/design/README.md
Comment (jonmeow) [inline docs/design/README.md]: If you're starting to add this, should it also be on other sections? e.g., for floating point types above, there isn't a backing proposal either, is there? I was specifically looking for bfloat16 and couldn't find any reference, although I see it's used by eigen3 and tensorflow (but how much else?). float80 is another curious choice to commit to -- who's using that?

Comment (josh11b) [reply docs/design/README.md]: I am in the process of making that change. It was unclear to me if I should be adding the provisional marks to this section while I'm touching it, or do all the provisional marks in a PR dedicated just to that.

Comment (jonmeow) [reply docs/design/README.md]: My own suggestion would be to try to add them comprehensively, rather than piecemeal. Having them in a few spots implies the rest _is_ firmed up.

Comment (josh11b) [reply docs/design/README.md]: Okay, will do. Just to note, I had a similar concern at a smaller scale here. Since I'm adding references, I was worried that strings themselves might look more finished than they are.

File: docs/design/expressions/implicit_conversions.md
Comment (chandlerc) [inline docs/design/expressions/implicit_conversions.md]: I find it easier to read this using the short type names. Is there a motivation to documenting with the long ones here?

File: docs/design/README.md
Comment (chandlerc) [reply docs/design/README.md]: I think this is a consequence of not having the broken-out files for everything.

Without those, we can't treat this file as comprehensively provisional, and just do the more nuanced documentation there. Is this having the fine grained provisionality going to be OK in order to allow us to mix provisional and decided things at this layer?

Comment (jonmeow) [reply docs/design/README.md]: I think that's kind of the state of things, that the design overview looks more finished than it is, not just for strings. I think I'm prejudiced though since I'd rather keep references in the individual designs, rather than copying some into the main readme.

File: docs/design/expressions/implicit_conversions.md
Comment (josh11b) [reply docs/design/expressions/implicit_conversions.md]: Do we allow the short type names for arbitrary `N` or just powers of 2?

File: docs/design/README.md
Comment (josh11b) [reply docs/design/README.md]: Looking at the deleted primitive_types.md, even the individual designs have the problem of mixing provisional and decided.

Comment (jonmeow) [reply docs/design/README.md]: In what way? It has a TODO at the top. I think when it was last updated, everything was provisional.

Comment (josh11b) [reply docs/design/README.md]: It started out completely provisional, but subsequent proposals have updated it.

File: docs/design/expressions/implicit_conversions.md
Comment (geoffromer) [inline docs/design/expressions/implicit_conversions.md:106]: Especially in places like this, this PR winds up giving the impression that the `iN` and `uN` types are aliases, but the `fN` types are not. I don't think that's our intent, so I think in contexts where we're talking about `Carbon.Int(N)`, we should talk about `Carbon.Float(N)`, not `fN`.

File: docs/design/README.md
Comment (geoffromer) [inline docs/design/README.md]: This makes it sound like `u8` is a distinct type from `Carbon.UInt(8)`. This is a pre-existing problem, but it seems within the scope of what this PR is trying to fix.

File: docs/design/README.md
Comment (josh11b) [reply docs/design/README.md]: Example: https://github.com/carbon-language/carbon-lang/pull/1901/files#diff-aea41795f576fecec3f8ba28d42efac82bcce528df637b06dc4ea7ddc1858a16

File: docs/design/expressions/implicit_conversions.md
Comment (chandlerc) [reply docs/design/expressions/implicit_conversions.md]: I somewhat hope they have the same restrictions, whatever those end up being.

I think the most problematic one for the short syntax will have to be covered either way: `i1`.

Comment (josh11b) [reply docs/design/expressions/implicit_conversions.md]: I'm going to put this PR on hold until #2015 is resolved.

Comment (josh11b) [reply docs/design/expressions/implicit_conversions.md]: I've switched this to follow the expected resolution from #2015

File: docs/design/expressions/implicit_conversions.md
Comment (josh11b) [reply docs/design/expressions/implicit_conversions.md:106]: I've undone this change, now that `iN` is legal for the same `N` as `Carbon.Int(N)`.

File: docs/design/README.md
Comment (josh11b) [reply docs/design/README.md]: I've reworded, is that better?

**Review by josh11b — COMMENTED** [review]
This should be ready as soon as #2015 is merged.

**Review by geoffromer — APPROVED** [review]

File: docs/design/expressions/implicit_conversions.md
Comment (josh11b) [reply docs/design/expressions/implicit_conversions.md]: #2015 has been merged so I'm resolving this.

**Pull request merged**
